---
title: "Recombinastics_analysis"
author: "Kenneth Matreyek"
date: "6/17/2020"
output: github_document
---

```{r Set up the workspace}
rm(list = ls())
library(tidyverse)
library(ggrepel)
library(reshape)
library(abind)
library(gridExtra)

theme_set(theme_bw())
theme_update(panel.grid.minor = element_blank())
```

## Look at the recombination method test dasta
```{r data import, fig.height = 4, fig.width = 6}
methods_data <- read.csv(file = "data/Recombination_method_tests.csv", header = T)

methods_data_replicates <- ncol(methods_data) - 5
methods_data$mean <- rowMeans(methods_data[,c("Recombined_Rep1","Recombined_Rep2","Recombined_Rep3","Recombined_Rep4")], na.rm = T)
methods_data$sd <- sqrt((methods_data$Recombined_Rep1 - methods_data$mean)^2 + 
                        (methods_data$Recombined_Rep2 - methods_data$mean)^2 + 
                        (methods_data$Recombined_Rep3 - methods_data$mean)^2 +
                        (methods_data$Recombined_Rep4 - methods_data$mean)^2) 
methods_data$se <- methods_data$sd / sqrt(methods_data_replicates - 1)

methods_data$upper_conf <- methods_data$mean + methods_data$se * 1.96
methods_data$lower_conf <- methods_data$mean - methods_data$se * 1.96

methods_data2 <- methods_data %>% filter(!is.na(Recombined_Rep1) & Method != "none") %>% mutate(concat = paste("Method: ",Method,"\n","Bxb1: ",Bxb1,sep = ""))
methods_data2$WellSize <- factor(methods_data2$WellSize)
methods_data2$concat <- factor(methods_data2$concat, levels = methods_data2$concat[c(1,3,2,4)])

methods_data2[methods_data2$lower_conf < 0,"lower_conf"] <- 0

Recombination_method_plot <- ggplot() + 
  theme_classic() + theme(axis.text.x = element_text(angle = -90, vjust = 0.5)) +
  scale_y_continuous(expand = c(0,0), limits = c(0,65)) +
  ylab("% mCherry positive cells") +
  xlab(NULL) +
  geom_hline(yintercept = 0) + 
  geom_errorbar(data = methods_data2, aes(x = concat, ymin = lower_conf, ymax = upper_conf, color = WellSize), 
                width = 0.2, position=position_dodge(width=0.3), size = 0.3 ) +
  geom_point(data = methods_data2, aes(x= concat, y=mean, color = WellSize), 
             position=position_dodge(width=0.3), shape = 95, size = 2) +
  geom_jitter(data = methods_data2, aes(x= concat, y=Recombined_Rep1, color = WellSize), 
             size = 0.9, position=position_dodge(width=0.3), alpha = 0.4) +
  geom_jitter(data = methods_data2, aes(x= concat, y=Recombined_Rep2, color = WellSize), 
             size = 0.9, position=position_dodge(width=0.3), alpha = 0.4) +
  geom_jitter(data = methods_data2, aes(x= concat, y=Recombined_Rep3, color = WellSize), 
             size = 0.9, position=position_dodge(width=0.3), alpha = 0.4) +
  geom_jitter(data = methods_data2, aes(x= concat, y=Recombined_Rep4, color = WellSize), 
             size = 0.9, position=position_dodge(width=0.3), alpha = 0.4)
print(Recombination_method_plot)
```


# This next chunk is for testing the orthogonality of the GT and GA Bxb1 sequences
```{r Stuff, fig.height = 2, fig.width = 3}
rep_1_frame <- read.csv(file = "data/GT_vs_GA_200206.csv", header = T, stringsAsFactors = F) %>% arrange(attp, bxb1, mix) %>% filter(bxb1 != "none")
rep_2_frame <- read.csv(file = "data/GT_vs_GA_200213.csv", header = T, stringsAsFactors = F) %>% arrange(attp, bxb1, mix) %>% filter(bxb1 != "none")
rep_3_frame <- read.csv(file = "data/GT_vs_GA_200219.csv", header = T, stringsAsFactors = F) %>% arrange(attp, bxb1, mix) %>% filter(bxb1 != "none")

replicates <- 2
rep_1 <- log10(rep_1_frame[,5:8])
rep_2 <- log10(rep_2_frame[,5:8])
rep_3 <- log10(rep_3_frame[,5:8])
means <- (rep_1 + rep_2 + rep_3)/3

#https://stackoverflow.com/questions/32609926/performing-element-wise-standard-deviation-in-r-with-two-matrices
m <- abind(rep_1, rep_2, rep_3, along=3)
standard_devs <- data.frame(apply(m, 1:2, sd))
standard_errors <-  standard_devs / sqrt(replicates)

upper_conf <- means + standard_errors * 1.96
lower_conf <- means - standard_errors * 1.96

label_frame <- rep_1_frame[,1:4] %>% mutate(comboname = paste(bxb1, mix, sep = "\n"))

means2 <- cbind(label_frame,means)
upper_conf2 <- cbind(label_frame,upper_conf)
lower_conf2 <- cbind(label_frame,lower_conf)
#ga_data_filtered$attp <- factor(ga_data_filtered$attp, levels = c("gt","ga"))

upper_conf2_melted <- melt(upper_conf2[,c("comboname","attp","gfp","mcherry")], by = c("comboname","attp"))
lower_conf2_melted <- melt(lower_conf2[,c("comboname","attp","gfp","mcherry")], by = c("comboname","attp"))
means2_melted <- melt(means2[,c("comboname","attp","gfp","mcherry")], by = c("comboname","attp"))


green_red_colorscale <- c(gfp = "green", mcherry = "red")

GT_vs_GA_plot <- ggplot() + 
  theme_classic() + theme(axis.text.x = element_text(angle = -90, vjust = 0.5)) +
  ylab("% cells recombined") +
  xlab(NULL) +
  scale_y_log10() +
  scale_color_manual(values = green_red_colorscale) + 
  geom_errorbar(data = upper_conf2_melted, aes(x = comboname, color = variable, ymin = 10^lower_conf2_melted$value, ymax = 10^value),
                alpha = 0.5, width = 0.2, position = position_dodge(width = 0.5)) +
  geom_point(data = means2_melted, aes(x = comboname, y = 10^value, color = variable), alpha = 0.5, position = position_dodge(width = 0.5)) +
  facet_wrap(~attp)
print(GT_vs_GA_plot)

ggsave(file = "Plots/GT_vs_GA_plot.pdf", GT_vs_GA_plot, height = 1.8, width = 3.8)
```


```{r Diving into clone2 data}
c2_none_raw <- read.csv(file = "data/flow/201002_F69_G783A_Clone_Comparisons/Clone2_none.csv")
c2_none <- c2_none_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c2_none) <- c("fsc","ssc","blu","grn","red","nir")
c2_none$clone <- "c2"; c2_none$treatment <- "none"
c2_recomb_raw <- read.csv(file = "data/flow/201002_F69_G783A_Clone_Comparisons/Clone2_recomb.csv")
c2_recomb <- c2_recomb_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c2_recomb) <- c("fsc","ssc","blu","grn","red","nir")
c2_recomb$clone <- "c2"; c2_recomb$treatment <- "recomb"
c2_ap1903_raw <- read.csv(file = "data/flow/201002_F69_G783A_Clone_Comparisons/Clone2_ap1903.csv")
c2_ap1903 <- c2_ap1903_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c2_ap1903) <- c("fsc","ssc","blu","grn","red","nir")
c2_ap1903$clone <- "c2"; c2_ap1903$treatment <- "ap1903"


cell_number <- 2000
combined_data <- rbind(c2_none[1:cell_number,], c2_recomb[1:cell_number,], c2_ap1903[1:cell_number,])
combined_data$treatment <- factor(combined_data$treatment, levels = c("recomb","none","ap1903"))

plot_alpha <- 0.2
axis_limits <- c(10,1e6)

custom_color_scale <- c("none" = "magenta", "recomb" = "black", "ap1903" = "cyan")

c2_bn_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = nir, color = treatment), alpha = plot_alpha)

c2_bg_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = grn, color = treatment), alpha = plot_alpha)

c2_br_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = red, color = treatment), alpha = plot_alpha)

c2_ng_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = nir, y = grn, color = treatment), alpha = plot_alpha)

c2_nr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = nir, y = red, color = treatment), alpha = plot_alpha)

c2_gr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = grn, y = red, color = treatment), alpha = plot_alpha) +
  geom_rect(mapping = aes(xmin = 3e4, xmax = 1e6, ymin = 3e4, ymax = 1e6), alpha = 0, color = "black", linetype = 2)
c2_gr_plot

c2_arranged_plot <- grid.arrange(c2_bn_plot, c2_bg_plot, c2_br_plot, c2_ng_plot, c2_nr_plot, c2_gr_plot, ncol=6, nrow=1)

ggsave(file = "plots/201002/c2_arranged_plot.png", c2_arranged_plot, height = 3, width = 18)

paste("Percent double positive before selection:", round(sum(c2_recomb$grn >= 3e4 & c2_recomb$red >= 3e4) / nrow(c2_recomb) * 100,2))
paste("Percent double positive after selection:", round(sum(c2_ap1903$grn >= 3e4 & c2_ap1903$red >= 3e4) / nrow(c2_ap1903) * 100,2))
```

```{r Diving into clone2 data}
c6_none_raw <- read.csv(file = "data/flow/201002_F69_G783A_Clone_Comparisons/Clone6_none.csv")
c6_none <- c6_none_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c6_none) <- c("fsc","ssc","blu","grn","red","nir")
c6_none$clone <- "c6"; c6_none$treatment <- "none"
c6_recomb_raw <- read.csv(file = "data/flow/201002_F69_G783A_Clone_Comparisons/Clone6_recomb.csv")
c6_recomb <- c6_recomb_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c6_recomb) <- c("fsc","ssc","blu","grn","red","nir")
c6_recomb$clone <- "c6"; c6_recomb$treatment <- "recomb"
c6_ap1903_raw <- read.csv(file = "data/flow/201002_F69_G783A_Clone_Comparisons/Clone6_ap1903.csv")
c6_ap1903 <- c6_ap1903_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c6_ap1903) <- c("fsc","ssc","blu","grn","red","nir")
c6_ap1903$clone <- "c6"; c6_ap1903$treatment <- "ap1903"


cell_number <- 2000
combined_data <- rbind(c6_none[1:cell_number,], c6_recomb[1:cell_number,], c6_ap1903[1:cell_number,])
combined_data$treatment <- factor(combined_data$treatment, levels = c("recomb","none","ap1903"))

plot_alpha <- 0.2
axis_limits <- c(10,1e6)

custom_color_scale <- c("none" = "magenta", "recomb" = "black", "ap1903" = "cyan")

c6_bn_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = nir, color = treatment), alpha = plot_alpha)

c6_bg_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = grn, color = treatment), alpha = plot_alpha)

c6_br_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = red, color = treatment), alpha = plot_alpha)

c6_ng_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = nir, y = grn, color = treatment), alpha = plot_alpha)

c6_nr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = nir, y = red, color = treatment), alpha = plot_alpha)

c6_gr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = grn, y = red, color = treatment), alpha = plot_alpha) +
  geom_rect(mapping = aes(xmin = 3e4, xmax = 1e6, ymin = 3e4, ymax = 1e6), alpha = 0, color = "black", linetype = 2)
c6_gr_plot

c6_arranged_plot <- grid.arrange(c6_bn_plot, c6_bg_plot, c6_br_plot, c6_ng_plot, c6_nr_plot, c6_gr_plot, ncol=6, nrow=1)

ggsave(file = "plots/201002/c6_arranged_plot.png", c6_arranged_plot, height = 3, width = 18)

paste("Percent double positive before selection:", round(sum(c6_recomb$grn >= 3e4 & c6_recomb$red >= 3e4) / nrow(c6_recomb) * 100,2))
paste("Percent double positive after selection:", round(sum(c6_ap1903$grn >= 3e4 & c6_ap1903$red >= 3e4) / nrow(c6_ap1903) * 100,2))
```

```{r Diving into clone11 data}
c11_none_raw <- read.csv(file = "data/flow/201002_F69_G783A_Clone_Comparisons/Clone11_none.csv")
c11_none <- c11_none_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c11_none) <- c("fsc","ssc","blu","grn","red","nir")
c11_none$clone <- "c11"; c11_none$treatment <- "none"
c11_recomb_raw <- read.csv(file = "data/flow/201002_F69_G783A_Clone_Comparisons/Clone11_recomb.csv")
c11_recomb <- c11_recomb_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c11_recomb) <- c("fsc","ssc","blu","grn","red","nir")
c11_recomb$clone <- "c11"; c11_recomb$treatment <- "recomb"
c11_ap1903_raw <- read.csv(file = "data/flow/201002_F69_G783A_Clone_Comparisons/Clone11_ap1903.csv")
c11_ap1903 <- c11_ap1903_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c11_ap1903) <- c("fsc","ssc","blu","grn","red","nir")
c11_ap1903$clone <- "c11"; c11_ap1903$treatment <- "ap1903"


cell_number <- 2000
combined_data <- rbind(c11_none[1:cell_number,], c11_recomb[1:cell_number,], c11_ap1903[1:cell_number,])
combined_data$treatment <- factor(combined_data$treatment, levels = c("recomb","none","ap1903"))

plot_alpha <- 0.2
axis_limits <- c(10,1e6)

custom_color_scale <- c("none" = "magenta", "recomb" = "black", "ap1903" = "cyan")

c11_bn_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = nir, color = treatment), alpha = plot_alpha)

c11_bg_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = grn, color = treatment), alpha = plot_alpha)

c11_br_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = red, color = treatment), alpha = plot_alpha)

c11_ng_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = nir, y = grn, color = treatment), alpha = plot_alpha)

c11_nr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = nir, y = red, color = treatment), alpha = plot_alpha)

c11_gr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = grn, y = red, color = treatment), alpha = plot_alpha) +
  geom_rect(mapping = aes(xmin = 3e4, xmax = 1e6, ymin = 3e4, ymax = 1e6), alpha = 0, color = "black", linetype = 2)
c11_gr_plot

c11_arranged_plot <- grid.arrange(c11_bn_plot, c11_bg_plot, c11_br_plot, c11_ng_plot, c11_nr_plot, c11_gr_plot, ncol=6, nrow=1)

ggsave(file = "plots/201002/c11_arranged_plot.png", c11_arranged_plot, height = 3, width = 18)

paste("Percent double positive before selection:", round(sum(c11_recomb$grn >= 3e4 & c11_recomb$red >= 3e4) / nrow(c11_recomb) * 100,2))
paste("Percent double positive after selection:", round(sum(c11_ap1903$grn >= 3e4 & c11_ap1903$red >= 3e4) / nrow(c11_ap1903) * 100,2))
```




##### THIS IS NOW THE REPEAT SET OF DATA

```{r Diving into clone2 data}
c2_none_raw <- read.csv(file = "data/flow/201016_F74_G783A_Clone_Comparisons/Clone2_none.csv")
c2_none <- c2_none_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c2_none) <- c("fsc","ssc","blu","grn","red","nir")
c2_none$clone <- "c2"; c2_none$treatment <- "none"
c2_recomb_raw <- read.csv(file = "data/flow/201016_F74_G783A_Clone_Comparisons/Clone2_recomb.csv")
c2_recomb <- c2_recomb_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c2_recomb) <- c("fsc","ssc","blu","grn","red","nir")
c2_recomb$clone <- "c2"; c2_recomb$treatment <- "recomb"
c2_ap1903_raw <- read.csv(file = "data/flow/201016_F74_G783A_Clone_Comparisons/Clone2_ap1903.csv")
c2_ap1903 <- c2_ap1903_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c2_ap1903) <- c("fsc","ssc","blu","grn","red","nir")
c2_ap1903$clone <- "c2"; c2_ap1903$treatment <- "ap1903"


cell_number <- 2000
combined_data <- rbind(c2_none[1:cell_number,], c2_recomb[1:cell_number,], c2_ap1903[1:cell_number,])
combined_data$treatment <- factor(combined_data$treatment, levels = c("recomb","none","ap1903"))

plot_alpha <- 0.2
axis_limits <- c(10,1e6)

custom_color_scale <- c("none" = "magenta", "recomb" = "black", "ap1903" = "cyan")

c2_bn_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = nir, color = treatment), alpha = plot_alpha)

c2_bg_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = grn, color = treatment), alpha = plot_alpha)

c2_br_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = red, color = treatment), alpha = plot_alpha)

c2_ng_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = nir, y = grn, color = treatment), alpha = plot_alpha)

c2_nr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = nir, y = red, color = treatment), alpha = plot_alpha)

c2_gr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = grn, y = red, color = treatment), alpha = plot_alpha) +
  geom_rect(mapping = aes(xmin = 3e4, xmax = 1e6, ymin = 3e4, ymax = 1e6), alpha = 0, color = "black", linetype = 2)
c2_gr_plot

c2_arranged_plot <- grid.arrange(c2_bn_plot, c2_bg_plot, c2_br_plot, c2_ng_plot, c2_nr_plot, c2_gr_plot, ncol=6, nrow=1)

ggsave(file = "plots/201016/c2_arranged_plot.png", c2_arranged_plot, height = 3, width = 18)

paste("Percent double positive before selection:", round(sum(c2_recomb$grn >= 3e4 & c2_recomb$red >= 3e4) / nrow(c2_recomb) * 100,2))
paste("Percent double positive after selection:", round(sum(c2_ap1903$grn >= 3e4 & c2_ap1903$red >= 3e4) / nrow(c2_ap1903) * 100,2))
```

```{r Diving into clone2 data}
c6_none_raw <- read.csv(file = "data/flow/201016_F74_G783A_Clone_Comparisons/Clone6_none.csv")
c6_none <- c6_none_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c6_none) <- c("fsc","ssc","blu","grn","red","nir")
c6_none$clone <- "c6"; c6_none$treatment <- "none"
c6_recomb_raw <- read.csv(file = "data/flow/201016_F74_G783A_Clone_Comparisons/Clone6_recomb.csv")
c6_recomb <- c6_recomb_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c6_recomb) <- c("fsc","ssc","blu","grn","red","nir")
c6_recomb$clone <- "c6"; c6_recomb$treatment <- "recomb"
c6_ap1903_raw <- read.csv(file = "data/flow/201016_F74_G783A_Clone_Comparisons/Clone6_ap1903.csv")
c6_ap1903 <- c6_ap1903_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c6_ap1903) <- c("fsc","ssc","blu","grn","red","nir")
c6_ap1903$clone <- "c6"; c6_ap1903$treatment <- "ap1903"


cell_number <- 2000
combined_data <- rbind(c6_none[1:cell_number,], c6_recomb[1:cell_number,], c6_ap1903[1:cell_number,])
combined_data$treatment <- factor(combined_data$treatment, levels = c("recomb","none","ap1903"))

plot_alpha <- 0.2
axis_limits <- c(10,1e6)

custom_color_scale <- c("none" = "magenta", "recomb" = "black", "ap1903" = "cyan")

c6_bn_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = nir, color = treatment), alpha = plot_alpha)

c6_bg_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = grn, color = treatment), alpha = plot_alpha)

c6_br_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = red, color = treatment), alpha = plot_alpha)

c6_ng_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = nir, y = grn, color = treatment), alpha = plot_alpha)

c6_nr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = nir, y = red, color = treatment), alpha = plot_alpha)

c6_gr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = grn, y = red, color = treatment), alpha = plot_alpha) +
  geom_rect(mapping = aes(xmin = 3e4, xmax = 1e6, ymin = 3e4, ymax = 1e6), alpha = 0, color = "black", linetype = 2)
c6_gr_plot

c6_arranged_plot <- grid.arrange(c6_bn_plot, c6_bg_plot, c6_br_plot, c6_ng_plot, c6_nr_plot, c6_gr_plot, ncol=6, nrow=1)

ggsave(file = "plots/201016/c6_arranged_plot.png", c6_arranged_plot, height = 3, width = 18)

paste("Percent double positive before selection:", round(sum(c6_recomb$grn >= 3e4 & c6_recomb$red >= 3e4) / nrow(c6_recomb) * 100,2))
paste("Percent double positive after selection:", round(sum(c6_ap1903$grn >= 3e4 & c6_ap1903$red >= 3e4) / nrow(c6_ap1903) * 100,2))
```

```{r Diving into clone11 data}
c11_none_raw <- read.csv(file = "data/flow/201016_F74_G783A_Clone_Comparisons/Clone11_none.csv")
c11_none <- c11_none_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c11_none) <- c("fsc","ssc","blu","grn","red","nir")
c11_none$clone <- "c11"; c11_none$treatment <- "none"
c11_recomb_raw <- read.csv(file = "data/flow/201016_F74_G783A_Clone_Comparisons/Clone11_recomb.csv")
c11_recomb <- c11_recomb_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c11_recomb) <- c("fsc","ssc","blu","grn","red","nir")
c11_recomb$clone <- "c11"; c11_recomb$treatment <- "recomb"
c11_ap1903_raw <- read.csv(file = "data/flow/201016_F74_G783A_Clone_Comparisons/Clone11_ap1903.csv")
c11_ap1903 <- c11_ap1903_raw[,c("FSC.A","SSC.A","VL1.A","BL1.A","YL2.A","RL1.A")]; colnames(c11_ap1903) <- c("fsc","ssc","blu","grn","red","nir")
c11_ap1903$clone <- "c11"; c11_ap1903$treatment <- "ap1903"


cell_number <- 2000
combined_data <- rbind(c11_none[1:cell_number,], c11_recomb[1:cell_number,], c11_ap1903[1:cell_number,])
combined_data$treatment <- factor(combined_data$treatment, levels = c("recomb","none","ap1903"))

plot_alpha <- 0.2
axis_limits <- c(10,1e6)

custom_color_scale <- c("none" = "magenta", "recomb" = "black", "ap1903" = "cyan")

c11_bn_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = nir, color = treatment), alpha = plot_alpha)

c11_bg_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = grn, color = treatment), alpha = plot_alpha)

c11_br_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = blu, y = red, color = treatment), alpha = plot_alpha)

c11_ng_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = nir, y = grn, color = treatment), alpha = plot_alpha)

c11_nr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = nir, y = red, color = treatment), alpha = plot_alpha)

c11_gr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data, aes(x = grn, y = red, color = treatment), alpha = plot_alpha) +
  geom_rect(mapping = aes(xmin = 3e4, xmax = 1e6, ymin = 3e4, ymax = 1e6), alpha = 0, color = "black", linetype = 2)
c11_gr_plot

c11_arranged_plot <- grid.arrange(c11_bn_plot, c11_bg_plot, c11_br_plot, c11_ng_plot, c11_nr_plot, c11_gr_plot, ncol=6, nrow=1)

ggsave(file = "plots/201016/c11_arranged_plot.png", c11_arranged_plot, height = 3, width = 18)

paste("Percent double positive before selection:", round(sum(c11_recomb$grn >= 3e4 & c11_recomb$red >= 3e4) / nrow(c11_recomb) * 100,2))
paste("Percent double positive after selection:", round(sum(c11_ap1903$grn >= 3e4 & c11_ap1903$red >= 3e4) / nrow(c11_ap1903) * 100,2))
```


```{r Clone11 plots now for paper}

plot_alpha <- 0.2
axis_limits <- c(10,1e6)

c11_none_bn_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data %>% filter(treatment == "none"), aes(x = blu, y = nir), alpha = plot_alpha)
c11_none_bn_plot

c11_recomb_bn_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "recomb") + 
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data %>% filter(treatment == "recomb"), aes(x = blu, y = nir), alpha = plot_alpha)
c11_recomb_bn_plot

c11_ap1903_bn_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "ap1903") + 
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data %>% filter(treatment == "ap1903"), aes(x = blu, y = nir), alpha = plot_alpha)
c11_ap1903_bn_plot

c11_none_gr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data %>% filter(treatment == "none"), aes(x = grn, y = red), alpha = plot_alpha)
c11_none_gr_plot

c11_recomb_gr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "recomb") + 
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data %>% filter(treatment == "recomb"), aes(x = grn, y = red), alpha = plot_alpha)
c11_recomb_gr_plot

c11_ap1903_gr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "ap1903") + 
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = combined_data %>% filter(treatment == "ap1903"), aes(x = grn, y = red), alpha = plot_alpha)
c11_ap1903_gr_plot


c11_arranged_plot <- grid.arrange(c11_bn_plot, c11_bg_plot, c11_br_plot, c11_ng_plot, c11_nr_plot, c11_gr_plot, ncol=6, nrow=1)

ggsave(file = "plots/201016/c11_arranged_plot.png", c11_arranged_plot, height = 3, width = 18)

paste("Percent double positive before selection:", round(sum(c11_recomb$grn >= 3e4 & c11_recomb$red >= 3e4) / nrow(c11_recomb) * 100,2))
paste("Percent double positive after selection:", round(sum(c11_ap1903$grn >= 3e4 & c11_ap1903$red >= 3e4) / nrow(c11_ap1903) * 100,2))






```

## TRYING TO PLOT THE DOUBLE LANDING PAD COLORS

```{r Principal component analysis}
# http://www.sthda.com/english/articles/31-principal-component-methods-in-r-practical-guide/118-principal-component-analysis-in-r-prcomp-vs-princomp/

library(factoextra)
library(ggfortify)

combined_data <- rbind(c11_none[1:2000,], c11_recomb[1:2000,], c11_ap1903[1:2000,])

ggplot() + theme_bw() + scale_x_log10() + scale_y_log10() + geom_point(data = combined_data, aes(x = red, y = nir))

combined_data$zir <- combined_data$nir - combined_data$red * 0.01
ggplot() + theme_bw() + scale_x_log10() + scale_y_log10() + geom_point(data = combined_data, aes(x = red, y = zir))

combined_for_pca <- combined_data[,c("blu","grn","red","zir")]

clone2.pca <- prcomp(combined_for_pca, center = TRUE,scale. = TRUE)
fviz_eig(clone2.pca)

autoplot(clone2.pca, alpha = 0.2) +
  scale_x_continuous(limits = c(-0.3,0.3)) + scale_y_continuous(limits = c(-0.3,0.3))

fviz_pca_var(clone2.pca,
             col.var = "contrib", # Color by contributions to the PC
             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"),
             repel = TRUE     # Avoid text overlapping
             )

clone2.pca$rotation

pca2 <- data.frame("pc1" = clone2.pca$x[,1], "pc2" = clone2.pca$x[,2])

pca2$treatment <- combined_data[,"treatment"]

pca2$treatment <- factor(pca2$treatment, levels = c("none","recomb","ap1903"))
custom_color_scale <- c("none" = "magenta", "ap1903" = "cyan", "recomb" = "green")

ggplot() + theme_bw() + theme(panel.grid.minor = element_blank()) + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_continuous(limits = c(-5,5)) +
  scale_y_continuous(limits = c(-3,2)) +
  geom_point(data = pca2, aes(x = pc1, y = pc2, color = treatment), alpha = 0.05)

ggplot() + theme_bw() + theme(panel.grid.minor = element_blank()) + 
  scale_color_manual(values = custom_color_scale) +
  scale_x_continuous(limits = c(-5,5)) +
  scale_y_continuous(limits = c(-3,2)) +
  geom_point(data = pca2, aes(x = pc1, y = pc2), alpha = 0.05) +
  facet_grid(rows = vars(treatment))

pca_details <- cbind(combined_for_pca,pca2)

pca_details_melted <- melt(pca_details[,c("treatment","blu","grn","red","zir")], id. = "treatment")

ggplot() + theme_bw() +
  scale_x_log10() + 
  geom_histogram(data = pca_details_melted, aes(x = value)) +
  facet_grid(cols = vars(variable))

blu_cutoff <- 1e3
grn_cutoff <- 1e4
red_cutoff <- 1e4
zir_cutoff <- 1e3

ggplot() + theme_bw() + scale_x_log10() + geom_histogram(data = pca_details, aes(x = blu)) + geom_vline(xintercept = blu_cutoff)
ggplot() + theme_bw() + scale_x_log10() + geom_histogram(data = pca_details, aes(x = grn)) + geom_vline(xintercept = grn_cutoff)
ggplot() + theme_bw() + scale_x_log10() + geom_histogram(data = pca_details, aes(x = red)) + geom_vline(xintercept = red_cutoff)
ggplot() + theme_bw() + scale_x_log10() + geom_histogram(data = pca_details, aes(x = zir)) + geom_vline(xintercept = zir_cutoff)

pca_details$blu2 <- 0
pca_details$grn2 <- 0
pca_details$red2 <- 0
pca_details$zir2 <- 0

for(x in 1:nrow(pca_details)){
  if(pca_details$blu[x] > blu_cutoff){pca_details$blu2[x] <- 1}
  if(pca_details$grn[x] > grn_cutoff){pca_details$grn2[x] <- 1}
  if(pca_details$red[x] > red_cutoff){pca_details$red2[x] <- 1}
  if(pca_details$zir[x] > zir_cutoff){pca_details$zir2[x] <- 1}
}

pca_details$bgrn <- paste(pca_details$blu2,pca_details$grn2,pca_details$red2,pca_details$zir2,sep="")

bgrn_table <- data.frame(table(pca_details$bgrn)) %>% arrange(desc(Freq))

pca_details$label <- "Unannotated"

for(x in 1:nrow(pca_details)){
  if(pca_details$bgrn[x] == "1001"){pca_details$label[x] <- "Unrecombined"}
  if(pca_details$bgrn[x] == "0110"){pca_details$label[x] <- "Doubly_recombined"}
  if(pca_details$bgrn[x] == "1010"){pca_details$label[x] <- "GA_only"}
  if(pca_details$bgrn[x] == "0101"){pca_details$label[x] <- "GT_only"}
}

custom_color_scale2 <- c("Unannotated" = "black", "Unrecombined" = "blue", "Doubly_recombined" = "purple", "GA_only" = "red", "GT_only" = "green")

Clone11_PCA_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank()) + 
  scale_color_manual(values = custom_color_scale2) +
  scale_x_continuous(limits = c(-5,5)) +
  scale_y_continuous(limits = c(-3,2)) +
  geom_point(data = pca_details, aes(x = pc1, y = pc2, color = label), alpha = 0.1, size = 0.5) +
  facet_grid(rows = vars(treatment))
Clone11_PCA_plot
ggsave(file = "plots/Clone11_PCA_plot.pdf", Clone11_PCA_plot, height = 3.4, width = 4.5)

```

```{r Individual flow plots}
plot_alpha = 0.02
point_size = 0.25

c11_none_bz_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = pca_details %>% filter(treatment == "none"), aes(x = blu, y = zir), alpha = plot_alpha, size = point_size)
c11_none_bz_plot

c11_recomb_bz_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = pca_details %>% filter(treatment == "recomb"), aes(x = blu, y = zir), alpha = plot_alpha, size = point_size)
c11_recomb_bz_plot

c11_ap1903_bz_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = pca_details %>% filter(treatment == "ap1903"), aes(x = blu, y = zir), alpha = plot_alpha, size = point_size)
c11_ap1903_bz_plot

c11_none_gr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "none") + 
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = pca_details %>% filter(treatment == "none"), aes(x = grn, y = red), alpha = plot_alpha, size = point_size)
c11_none_gr_plot

c11_recomb_gr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "recomb") + 
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = pca_details %>% filter(treatment == "recomb"), aes(x = grn, y = red), alpha = plot_alpha, size = point_size)
c11_recomb_gr_plot

c11_ap1903_gr_plot <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank(), legend.position = "ap1903") + 
  scale_x_log10(limits = axis_limits) + scale_y_log10(limits = axis_limits) +
  geom_point(data = pca_details %>% filter(treatment == "ap1903"), aes(x = grn, y = red), alpha = plot_alpha, size = point_size)
c11_ap1903_gr_plot

c11_arranged_plot <- grid.arrange(c11_none_bz_plot, c11_recomb_bz_plot, c11_ap1903_bz_plot, c11_none_gr_plot, c11_recomb_gr_plot, c11_ap1903_gr_plot, ncol=3, nrow=2)

ggsave(file = "plots/c11_arranged_plot.pdf", c11_arranged_plot, height = 2.2, width = 4.1)

#paste("Percent double positive before selection:", round(sum(c11_recomb$grn >= 3e4 & c11_recomb$red >= 3e4) / nrow(c11_recomb) * 100,2))
#paste("Percent double positive after selection:", round(sum(c11_ap1903$grn >= 3e4 & c11_ap1903$red >= 3e4) / nrow(c11_ap1903) * 100,2))

pca_vectorplot <- fviz_pca_var(clone2.pca,
             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"),
             repel = TRUE     # Avoid text overlapping
             )
pca_vectorplot

ggsave(file = "plots/pca_vectorplot.pdf", pca_vectorplot, height = 2, width = 2)
```


```{r Flanking data - Flow Cyometry}
## Making a data frame to keep track of how many unexcised cells are observed with G718A
flanking_flow_df <- data.frame("flow" = NA, "frac_unexcised" = NA)
flank_red_pos_cutoff <- 3e3

## F46 LSR2
f46_g718a <- read.csv(file = "data/flow/Flanking/LSR2/F46_G718A_029_Singlets.csv") %>% mutate("flow" = "F46", sample = "Flanked")
f46_g747a <- read.csv(file = "data/flow/Flanking/LSR2/F46_G747A_030_Singlets.csv") %>% mutate("flow" = "F46", sample = "Control")

flank_expt_cell_num <- 12000
flank_red_pos_cutoff <- 1e4

f46 <- rbind(f46_g718a[1:flank_expt_cell_num,], f46_g747a[1:flank_expt_cell_num,])

ggplot() + theme_bw() + 
  scale_x_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  scale_y_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  geom_vline(xintercept = flank_red_pos_cutoff, linetype = 2) +
  geom_point(data = f46, aes(x = YG610.A, y = B525.A), alpha = 0.01) +
  facet_grid(rows = vars(sample))

f46$ratio <- f46$B525.A / f46$YG610.A
f46_subset <- f46 %>% filter(YG610.A >= flank_red_pos_cutoff)

f46_control_95pct_interval <- c(quantile((f46_subset %>% filter(sample == "Control"))$ratio,0.025),quantile((f46_subset %>% filter(sample == "Control"))$ratio,0.975))

ggplot() + theme_bw() + scale_x_continuous(limits = c(-0.05, 0.5)) +
  geom_histogram(data = f46_subset, aes(x = ratio), binwidth = 0.01) +
  facet_grid(rows = vars(sample)) +
  geom_vline(xintercept  = f46_control_95pct_interval, linetype = 2)

f46_fraction_unexcised <- sum((f46_subset %>% filter(sample == "Flanked"))$ratio > f46_control_95pct_interval[1])/nrow((f46_subset %>% filter(sample == "Flanked")))
flanking_flow_df <- rbind(flanking_flow_df, data.frame("flow" = "F46", "frac_unexcised" = f46_fraction_unexcised))

## F84
f84_none <- read.csv(file = "data/flow/Flanking/F84_None_Sample(1)_Singlets.csv") %>% mutate("flow" = "F84", sample = "None")
f84_g718a <- read.csv(file = "data/flow/Flanking/F84_G718A_Sample(4)_Singlets.csv") %>% mutate("flow" = "F84", sample = "Flanked")
f84_g747a <- read.csv(file = "data/flow/Flanking/F84_G747A_Sample(5)_Singlets.csv") %>% mutate("flow" = "F84", sample = "Control")

flank_expt_cell_num <- 90000
flank_red_pos_cutoff <- 5e3

f84 <- rbind(f84_none[1:flank_expt_cell_num,], f84_g718a[1:flank_expt_cell_num,], f84_g747a[1:flank_expt_cell_num,])

ggplot() + theme_bw() + 
  scale_x_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  scale_y_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  geom_vline(xintercept = flank_red_pos_cutoff, linetype = 2) +
  geom_point(data = f84, aes(x = YL2.A, y = BL1.A), alpha = 0.01) +
  facet_grid(rows = vars(sample))

f84$ratio <- f84$BL1.A / f84$YL2.A
f84_subset <- f84 %>% filter(YL2.A >= flank_red_pos_cutoff)

f84_control_95pct_interval <- c(quantile((f84_subset %>% filter(sample == "Control"))$ratio,0.025),quantile((f84_subset %>% filter(sample == "Control"))$ratio,0.975))

ggplot() + theme_bw() + scale_x_continuous(limits = c(-0.05, 0.5)) +
  geom_histogram(data = f84_subset, aes(x = ratio), binwidth = 0.01) +
  facet_grid(rows = vars(sample)) +
  geom_vline(xintercept  = f84_control_95pct_interval, linetype = 2)

f84_fraction_unexcised <- sum((f84_subset %>% filter(sample == "Flanked"))$ratio > f84_control_95pct_interval[1])/nrow((f84_subset %>% filter(sample == "Flanked")))
flanking_flow_df <- rbind(flanking_flow_df, data.frame("flow" = "F84", "frac_unexcised" = f84_fraction_unexcised))


## F101 Aria
f101_none <- read.csv(file = "data/flow/Flanking/Aria/F101_None_001_Singlets.csv") %>% mutate("flow" = "F101", sample = "None")
f101_g718a <- read.csv(file = "data/flow/Flanking/Aria/F101_G718A_002_Singlets.csv") %>% mutate("flow" = "F101", sample = "Flanked")
f101_g747a <- read.csv(file = "data/flow/Flanking/Aria/F101_G747A_003_Singlets.csv") %>% mutate("flow" = "F101", sample = "Control")

flank_expt_cell_num <- 2400
flank_red_pos_cutoff <- 5e3

f101 <- rbind(f101_none[1:flank_expt_cell_num,], f101_g718a[1:flank_expt_cell_num,], f101_g747a[1:flank_expt_cell_num,])

ggplot() + theme_bw() + 
  scale_x_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  scale_y_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  geom_vline(xintercept = flank_red_pos_cutoff, linetype = 2) +
  geom_point(data = f101, aes(x = G610.A, y = B515.A), alpha = 0.01) +
  facet_grid(rows = vars(sample))

f101$ratio <- f101$B515.A / f101$G610.A
f101_subset <- f101 %>% filter(G610.A >= flank_red_pos_cutoff)

f101_control_95pct_interval <- c(quantile((f101_subset %>% filter(sample == "Control"))$ratio,0.025),quantile((f101_subset %>% filter(sample == "Control"))$ratio,0.975))

ggplot() + theme_bw() + scale_x_continuous(limits = c(-0.05, 0.5)) +
  geom_histogram(data = f101_subset, aes(x = ratio), binwidth = 0.01) +
  facet_grid(rows = vars(sample)) +
  geom_vline(xintercept  = f101_control_95pct_interval, linetype = 2)

f101_fraction_unexcised <- sum((f101_subset %>% filter(sample == "Flanked"))$ratio > f101_control_95pct_interval[1])/nrow((f101_subset %>% filter(sample == "Flanked")))
flanking_flow_df <- rbind(flanking_flow_df, data.frame("flow" = "F101", "frac_unexcised" = f101_fraction_unexcised))


## F114
f114_g718a <- read.csv(file = "data/flow/Flanking/F114_G718A_A2_Single Cells.csv") %>% mutate("flow" = "F114", sample = "Flanked")
f114_g747a <- read.csv(file = "data/flow/Flanking/F114_G747A_A3_Single Cells.csv") %>% mutate("flow" = "F114", sample = "Control")

flank_expt_cell_num <- 45000
flank_red_pos_cutoff <- 3e3

f114 <- rbind(f114_g718a[1:flank_expt_cell_num,], f114_g747a[1:flank_expt_cell_num,])

ggplot() + theme_bw() + 
  scale_x_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  scale_y_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  geom_vline(xintercept = flank_red_pos_cutoff, linetype = 2) +
  geom_point(data = f114, aes(x = YL2.A, y = BL1.A), alpha = 0.1) +
  facet_grid(rows = vars(sample))

f114$ratio <- f114$BL1.A / f114$YL2.A
f114_subset <- f114 %>% filter(YL2.A >= flank_red_pos_cutoff)

f114_control_95pct_interval <- c(quantile((f114_subset %>% filter(sample == "Control"))$ratio,0.025),quantile((f114_subset %>% filter(sample == "Control"))$ratio,0.975))

ggplot() + theme_bw() + scale_x_continuous(limits = c(-0.05, 0.5)) +
  geom_histogram(data = f114_subset, aes(x = ratio), binwidth = 0.01) +
  facet_grid(rows = vars(sample)) +
  geom_vline(xintercept  = f114_control_95pct_interval, linetype = 2)

f114_fraction_unexcised <- sum((f114_subset %>% filter(sample == "Flanked"))$ratio > f114_control_95pct_interval[1])/nrow((f114_subset %>% filter(sample == "Flanked")))
flanking_flow_df <- rbind(flanking_flow_df, data.frame("flow" = "F114", "frac_unexcised" = f114_fraction_unexcised))


## F116
f116_g718a <- read.csv(file = "data/flow/Flanking/F116_G718A_H11_Singlets.csv") %>% mutate("flow" = "F116", sample = "Flanked")
f116_g747a <- read.csv(file = "data/flow/Flanking/F116_G747A_H12_Singlets.csv") %>% mutate("flow" = "F116", sample = "Control")
flank_expt_cell_num <- 25000
flank_red_pos_cutoff <- 5e3
f116 <- rbind(f116_g718a[1:flank_expt_cell_num,], f116_g747a[1:flank_expt_cell_num,])

ggplot() + theme_bw() + 
  scale_x_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  scale_y_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  geom_vline(xintercept = flank_red_pos_cutoff, linetype = 2) +
  geom_point(data = f116, aes(x = YL2.A, y = BL1.A), alpha = 0.1) +
  facet_grid(rows = vars(sample))

f116$ratio <- f116$BL1.A / f116$YL2.A
f116_subset <- f116 %>% filter(YL2.A >= flank_red_pos_cutoff)

f116_control_95pct_interval <- c(quantile((f116_subset %>% filter(sample == "Control"))$ratio,0.025),quantile((f116_subset %>% filter(sample == "Control"))$ratio,0.975))

ggplot() + theme_bw() + scale_x_continuous(limits = c(-0.05, 0.5)) +
  geom_histogram(data = f116_subset, aes(x = ratio), binwidth = 0.01) +
  facet_grid(rows = vars(sample)) +
  geom_vline(xintercept  = f116_control_95pct_interval, linetype = 2)

f116_fraction_unexcised <- sum((f116_subset %>% filter(sample == "Flanked"))$ratio > f116_control_95pct_interval[1])/nrow((f116_subset %>% filter(sample == "Flanked")))
flanking_flow_df <- rbind(flanking_flow_df, data.frame("flow" = "F116", "frac_unexcised" = f116_fraction_unexcised))


## F119
f119_none <- read.csv(file = "data/flow/Flanking/F119_None_G11_Singlets.csv") %>% mutate("flow" = "F119", sample = "None")
f119_g718a <- read.csv(file = "data/flow/Flanking/F119_G718A_H11_Singlets.csv") %>% mutate("flow" = "F119", sample = "Flanked")
f119_g747a <- read.csv(file = "data/flow/Flanking/F119_G747A_H12_Singlets.csv") %>% mutate("flow" = "F119", sample = "Control")
flank_expt_cell_num <- 9000
flank_red_pos_cutoff <- 5e3
f119 <- rbind(f119_none[1:flank_expt_cell_num,], f119_g718a[1:flank_expt_cell_num,], f119_g747a[1:flank_expt_cell_num,]) %>% filter(!is.na(sample))

ggplot() + theme_bw() + 
  scale_x_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  scale_y_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  geom_vline(xintercept = flank_red_pos_cutoff, linetype = 2) +
  geom_point(data = f119, aes(x = YL2.A, y = BL1.A), alpha = 0.1) +
  facet_grid(rows = vars(sample))

f119$ratio <- f119$BL1.A / f119$YL2.A
f119_subset <- f119 %>% filter(YL2.A >= flank_red_pos_cutoff)

f119_control_95pct_interval <- c(quantile((f119_subset %>% filter(sample == "Control"))$ratio,0.025),quantile((f119_subset %>% filter(sample == "Control"))$ratio,0.975))

ggplot() + theme_bw() + scale_x_continuous(limits = c(-0.05, 0.5)) +
  geom_histogram(data = f119_subset, aes(x = ratio), binwidth = 0.01) +
  facet_grid(rows = vars(sample)) +
  geom_vline(xintercept  = f119_control_95pct_interval, linetype = 2)

f119_fraction_unexcised <- sum((f119_subset %>% filter(sample == "Flanked"))$ratio > f119_control_95pct_interval[1])/nrow((f119_subset %>% filter(sample == "Flanked")))
flanking_flow_df <- rbind(flanking_flow_df, data.frame("flow" = "F119", "frac_unexcised" = f119_fraction_unexcised))


## F130
f130_none <- read.csv(file = "data/flow/Flanking/F130_none_D10_Singlets.csv") %>% mutate("flow" = "F130", sample = "None")
f130_g718a <- read.csv(file = "data/flow/Flanking/F130_G718A_D11_Singlets.csv") %>% mutate("flow" = "F130", sample = "Flanked")
f130_g747a <- read.csv(file = "data/flow/Flanking/F130_G747A_D12_Singlets.csv") %>% mutate("flow" = "F130", sample = "Control")
flank_expt_cell_num <- 89000
flank_red_pos_cutoff <- 5e3
f130 <- rbind(f130_none[1:flank_expt_cell_num,], f130_g718a[1:flank_expt_cell_num,], f130_g747a[1:flank_expt_cell_num,]) %>% filter(!is.na(sample))

ggplot() + theme_bw() + 
  scale_x_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  scale_y_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  geom_vline(xintercept = flank_red_pos_cutoff, linetype = 2) +
  geom_point(data = f130, aes(x = YL2.A, y = BL1.A), alpha = 0.1) +
  facet_grid(rows = vars(sample))

f130$ratio <- f130$BL1.A / f130$YL2.A
f130_subset <- f130 %>% filter(YL2.A >= flank_red_pos_cutoff)

f130_control_95pct_interval <- c(quantile((f130_subset %>% filter(sample == "Control"))$ratio,0.025),quantile((f130_subset %>% filter(sample == "Control"))$ratio,0.975))

ggplot() + theme_bw() + scale_x_continuous(limits = c(-0.05, 0.5)) +
  geom_histogram(data = f130_subset, aes(x = ratio), binwidth = 0.01) +
  facet_grid(rows = vars(sample)) +
  geom_vline(xintercept  = f130_control_95pct_interval, linetype = 2)

f130_fraction_unexcised <- sum((f130_subset %>% filter(sample == "Flanked"))$ratio > f130_control_95pct_interval[1])/nrow((f130_subset %>% filter(sample == "Flanked")))
flanking_flow_df <- rbind(flanking_flow_df, data.frame("flow" = "F130", "frac_unexcised" = f130_fraction_unexcised))


## F131
f131_none <- read.csv(file = "data/flow/Flanking/F131_none_D4_Singlets.csv") %>% mutate("flow" = "F131", sample = "None")
f131_g718a <- read.csv(file = "data/flow/Flanking/F131_G718A_D5_Singlets.csv") %>% mutate("flow" = "F131", sample = "Flanked")
f131_g747a <- read.csv(file = "data/flow/Flanking/F131_G747A_D6_Singlets.csv") %>% mutate("flow" = "F131", sample = "Control")
flank_expt_cell_num <- 190000
flank_red_pos_cutoff <- 3e3
f131 <- rbind(f131_none[1:flank_expt_cell_num,], f131_g718a[1:flank_expt_cell_num,], f131_g747a[1:flank_expt_cell_num,]) %>% filter(!is.na(sample))

ggplot() + theme_bw() + 
  scale_x_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  scale_y_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  geom_vline(xintercept = flank_red_pos_cutoff, linetype = 2) +
  geom_point(data = f131, aes(x = YL2.A, y = BL1.A), alpha = 0.1) +
  facet_grid(rows = vars(sample))

f131$ratio <- f131$BL1.A / f131$YL2.A
f131_subset <- f131 %>% filter(YL2.A >= flank_red_pos_cutoff)

f131_control_95pct_interval <- c(quantile((f131_subset %>% filter(sample == "Control"))$ratio,0.025),quantile((f131_subset %>% filter(sample == "Control"))$ratio,0.975))

ggplot() + theme_bw() + scale_x_continuous(limits = c(-0.05, 0.5)) +
  geom_histogram(data = f131_subset, aes(x = ratio), binwidth = 0.01) +
  facet_grid(rows = vars(sample)) +
  geom_vline(xintercept  = f131_control_95pct_interval, linetype = 2)

f131_fraction_unexcised <- sum((f131_subset %>% filter(sample == "Flanked"))$ratio > f131_control_95pct_interval[1])/nrow((f131_subset %>% filter(sample == "Flanked")))
flanking_flow_df <- rbind(flanking_flow_df, data.frame("flow" = "F131", "frac_unexcised" = f131_fraction_unexcised))


## F132
f132_g718a <- read.csv(file = "data/flow/Flanking/F132_G718A_G12_Singlets.csv") %>% mutate("flow" = "F132", sample = "Flanked")
f132_g747a <- read.csv(file = "data/flow/Flanking/F132_G747A_H1_Singlets.csv") %>% mutate("flow" = "F132", sample = "Control")
flank_expt_cell_num <- 180000
flank_red_pos_cutoff <- 3e3
f132 <- rbind(f132_g718a[1:flank_expt_cell_num,], f132_g747a[1:flank_expt_cell_num,])

ggplot() + theme_bw() + 
  scale_x_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  scale_y_log10(limits = c(1e1,1e5), expand = c(0,0)) + 
  geom_vline(xintercept = flank_red_pos_cutoff, linetype = 2) +
  geom_point(data = f132, aes(x = YL2.A, y = BL1.A), alpha = 0.1) +
  facet_grid(rows = vars(sample))

f132$ratio <- f132$BL1.A / f132$YL2.A
f132_subset <- f132 %>% filter(YL2.A >= flank_red_pos_cutoff)

f132_control_95pct_interval <- c(quantile((f132_subset %>% filter(sample == "Control"))$ratio,0.025),quantile((f132_subset %>% filter(sample == "Control"))$ratio,0.975))

ggplot() + theme_bw() + scale_x_continuous(limits = c(-0.05, 0.5)) +
  geom_histogram(data = f132_subset, aes(x = ratio), binwidth = 0.01) +
  facet_grid(rows = vars(sample)) +
  geom_vline(xintercept  = f132_control_95pct_interval, linetype = 2)

f132_fraction_unexcised <- sum((f132_subset %>% filter(sample == "Flanked"))$ratio > f132_control_95pct_interval[1])/nrow((f132_subset %>% filter(sample == "Flanked")))
flanking_flow_df <- rbind(flanking_flow_df, data.frame("flow" = "F132", "frac_unexcised" = f132_fraction_unexcised))
```


```{r Flanking data - Flow Cyometry}
## Some summary graphs
## An example set of samples
f131_example <- rbind(f131_none[1:20000,], f131_g718a[1:20000,], f131_g747a[1:20000,]) %>% filter(!is.na(sample))
f131_example$sample <- factor(f131_example$sample, levels = c("None", "Flanked", "Control"))
F131_example_scatterplot <- ggplot() + theme(panel.grid.minor = element_blank()) + 
  scale_x_log10(limits = c(1e1,1e5), expand = c(0,0), breaks = c(1e2,1e4)) + 
  scale_y_log10(limits = c(1e1,1e4), expand = c(0,0), breaks = c(1e2,1e3)) + 
  geom_point(data = f131_example, aes(x = YL2.A, y = BL1.A), alpha = 0.1, size = 0.5) +
  facet_grid(cols = vars(sample))
ggsave(file = "plots/Flanking_example_scatterplot.pdf", F131_example_scatterplot, height = 1.6, width = 4.5)
F131_example_scatterplot

## Looking at fraction excised over time
flanking_flow_days <- read.csv(file = "data/Flanking_expt_timepoints.csv", header = T, stringsAsFactors = F)
flanking_flow_df2 <- merge(flanking_flow_df, flanking_flow_days, by = "flow")

Flanking_timeplot <- ggplot() + theme(panel.grid.minor = element_blank()) + 
  scale_y_log10(limits = c(0.001,1)) + scale_x_log10() +
  labs(x = "Days after\ntransfection", y = "Fraction of\ncells unexcised") +
  geom_point(data = flanking_flow_df2 %>% filter(instrument == "Attune"), aes(x = days, y = frac_unexcised))
ggsave(file = "plots/Flanking_timeplot.pdf", Flanking_timeplot, height = 1.6, width = 2)
Flanking_timeplot

## Looking at the reproducibility of recombination
flank_indep_recombs <- flanking_flow_df2 %>% group_by(replicate) %>% summarize(frac_unexcised = 10^mean(log10(frac_unexcised)))
flank_flow_geomean <- 10^mean(log10(flank_indep_recombs$frac_unexcised))
flank_flow_upper_conf <- 10^(mean(log10(flank_indep_recombs$frac_unexcised)) + (sd(log10(flank_indep_recombs$frac_unexcised))/sqrt(nrow(flank_indep_recombs)) * 1.96))
flank_flow_lower_conf <- 10^(mean(log10(flank_indep_recombs$frac_unexcised)) - (sd(log10(flank_indep_recombs$frac_unexcised))/sqrt(nrow(flank_indep_recombs)) * 1.96))

Repeated_excisions_plot <- ggplot() + theme(panel.grid.minor = element_blank(), panel.grid.major.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  labs(x = NULL, y = "Fraction of cells unexcised") + 
  scale_y_log10(limits = c(0.001,1)) + #scale_x_continuous(limits = c(-0.1,0.1)) +
  geom_point(data = flank_indep_recombs, aes(x = 0, y = frac_unexcised), alpha = 0.5) +
  geom_point(aes(x = 0, y = flank_flow_geomean), shape = 95, size = 10) +
  geom_errorbar(aes(x = 0, ymax = flank_flow_upper_conf, ymin = flank_flow_lower_conf), width = 0.01)
ggsave(file = "plots/Repeated_excisions_plot.pdf", Repeated_excisions_plot, height = 1.1, width = 1)
Repeated_excisions_plot
```

```{r Flanking sequencing data}
flanked_seq <- read.csv(file = "data/Flanked_sequencing.csv", header = T, stringsAsFactors = F)

Flanking_sequencing <- ggplot() + theme(panel.grid.major.x = element_blank(), legend.position = "top") + 
  labs(x = NULL, y = "Fraction of reads") +
  geom_point(data = flanked_seq, aes(x = sample, y = fraction, color = result), position = position_dodge(width = 0.2))
ggsave(file = "plots/Flanking_sequencing.pdf", Flanking_sequencing, height = 1.8, width = 2.2)
Flanking_sequencing
```



## Transcriptional reporter
```{r Import the initial set of data showing how it works with EGF stimulation}
flow_01_none <- read.csv(file = "Data/EGF_stimulation_48hr/01_none.csv", header = T, stringsAsFactors = FALSE); colnames(flow_01_none) <- c("fsc_a","fsc_h","fsc_w","ssc_a","apc_a","gfp_a","red_a","blue_a");flow_01_none$order <- 1;flow_01_none <- flow_01_none[1:18000,]
flow_02_E <- read.csv(file = "Data/EGF_stimulation_48hr/02_E.csv", header = T, stringsAsFactors = FALSE); colnames(flow_02_E) <- c("fsc_a","fsc_h","fsc_w","ssc_a","apc_a","gfp_a","red_a","blue_a");flow_02_E$order <- 2;flow_02_E <- flow_02_E[1:18000,]
flow_03_T <- read.csv(file = "Data/EGF_stimulation_48hr/03_T.csv", header = T, stringsAsFactors = FALSE); colnames(flow_03_T) <- c("fsc_a","fsc_h","fsc_w","ssc_a","apc_a","gfp_a","red_a","blue_a");flow_03_T$order <- 3;flow_03_T <- flow_03_T[1:18000,]
flow_04_U <- read.csv(file = "Data/EGF_stimulation_48hr/04_U.csv", header = T, stringsAsFactors = FALSE); colnames(flow_04_U) <- c("fsc_a","fsc_h","fsc_w","ssc_a","apc_a","gfp_a","red_a","blue_a");flow_04_U$order <- 4;flow_04_U <- flow_04_U[1:18000,]
flow_05_ET <- read.csv(file = "Data/EGF_stimulation_48hr/05_ET.csv", header = T, stringsAsFactors = FALSE); colnames(flow_05_ET) <- c("fsc_a","fsc_h","fsc_w","ssc_a","apc_a","gfp_a","red_a","blue_a");flow_05_ET$order <- 5;flow_05_ET <- flow_05_ET[1:18000,]
flow_06_EU <- read.csv(file = "Data/EGF_stimulation_48hr/06_EU.csv", header = T, stringsAsFactors = FALSE); colnames(flow_06_EU) <- c("fsc_a","fsc_h","fsc_w","ssc_a","apc_a","gfp_a","red_a","blue_a");flow_06_EU$order <- 6;flow_06_EU <- flow_06_EU[1:18000,]
flow_07_TU <- read.csv(file = "Data/EGF_stimulation_48hr/07_TU.csv", header = T, stringsAsFactors = FALSE); colnames(flow_07_TU) <- c("fsc_a","fsc_h","fsc_w","ssc_a","apc_a","gfp_a","red_a","blue_a");flow_07_TU$order <- 7;flow_07_TU <- flow_07_TU[1:18000,]
flow_08_ETU <- read.csv(file = "Data/EGF_stimulation_48hr/08_ETU.csv", header = T, stringsAsFactors = FALSE); colnames(flow_08_ETU) <- c("fsc_a","fsc_h","fsc_w","ssc_a","apc_a","gfp_a","red_a","blue_a");flow_08_ETU$order <- 8;flow_08_ETU <- flow_08_ETU[1:18000,]

flow_combined <- rbind(flow_01_none, flow_02_E, flow_03_T, flow_04_U, flow_05_ET, flow_06_EU, flow_07_TU, flow_08_ETU)

flow_combined_summary <- flow_combined %>% group_by(order) %>% summarize(median = median(gfp_a))

flow_combined_summary$median[5] / flow_combined_summary$median[1]

EGF_stimulation_plot <- ggplot() + 
  xlab("Green fluorescence") +
  ylab("Number of cells") +
  scale_x_log10(limits = c(10,1e5), expand = c(0,0)) + 
  scale_y_continuous(breaks = c(250)) +
  geom_histogram(data = flow_combined, aes(x = gfp_a), bins = 100, alpha = 0.75) +
  facet_grid(rows = vars(order), scales = "free_y")
EGF_stimulation_plot
ggsave(file = "Plots/EGF_stimulation_plot.pdf", EGF_stimulation_plot, height = 60, width = 70, units = "mm")
```















```{r Recombination mixture experiments}
recomb_mixture <- read.csv(file = "data/recomb_mixture.csv", header = T, stringsAsFactors = F)

Orthogonal_plasmid_recombinations <- ggplot() + scale_y_log10() + theme(panel.grid.major.x = element_blank()) +
  labs(x = "AttP plasmid (individually tested)", y = "Fraction of\nrecombinants", color = "AttB\nplasmid\n(mixed)") + 
  geom_point(data = recomb_mixture, aes(x = landing_pad, y = fraction, color = recomb_vector), position = position_dodge(width = 0.4))
ggsave(file = "plots/Orthogonal_plasmid_recombinations.pdf", Orthogonal_plasmid_recombinations, height = 2, width = 4)
Orthogonal_plasmid_recombinations
```


```{r Recombination mixture experiments 2}
recomb_mixture_test <- recomb_mixture[1:16,c("landing_pad","recomb_vector")]
colnames(recomb_mixture_test) <- c("lp1","lp2")

simulation_test_size = 20000
recomb_mixture_test$on_target <- NA

for(x in 1:nrow(recomb_mixture_test)){
  recomb_mix_lp1 <- recomb_mixture_test$lp1[x]
  recomb_mix_lp2 <- recomb_mixture_test$lp2[x]
  if(recomb_mix_lp1 == recomb_mix_lp2){
    lp1_sampling <- sample(x = c(1,0), prob = c(0.5,0.5) , size = simulation_test_size, replace = T)
    lp2_sampling <- sample(x = c(1,0), prob = c(0.5,0.5) , size = simulation_test_size, replace = T)
    results_frame <- data.frame("lp1_on_target" = lp1_sampling == 1, "lp2_on_target" = lp2_sampling == 1)
    results_frame$sum <- results_frame$lp1_on_target + results_frame$lp2_on_target
    recomb_mixture_test$on_target[x] <- sum(results_frame$sum == 1)
  }
  if(recomb_mix_lp1 != recomb_mix_lp2){
    lp1_probs <- recomb_mixture %>% filter(landing_pad == recomb_mix_lp1 & recomb_vector %in% c(recomb_mix_lp1, recomb_mix_lp2))
    lp2_probs <- recomb_mixture %>% filter(landing_pad == recomb_mix_lp2 & recomb_vector %in% c(recomb_mix_lp1, recomb_mix_lp2))
    lp1_sampling <- sample(x = lp1_probs$recomb_vector, prob = lp1_probs$fraction , size = simulation_test_size, replace = T)
    lp2_sampling <- sample(x = lp2_probs$recomb_vector, prob = lp2_probs$fraction , size = simulation_test_size, replace = T)
    results_frame <- data.frame("lp1_on_target" = lp1_sampling == recomb_mix_lp1, "lp2_on_target" = lp2_sampling == recomb_mix_lp2)
    results_frame$sum <- results_frame$lp1_on_target + results_frame$lp2_on_target
    recomb_mixture_test$on_target[x] <- sum(results_frame$sum == 2)
  }
}
recomb_mixture_test$frac_on_target <- recomb_mixture_test$on_target / simulation_test_size

Double_LP_heatmap <- ggplot() + labs(x = "First landing pad", y = "Second landing pad", fill = "Fraction\nwith unique\nrecombinant\nplasmid pair") + 
  scale_fill_gradient2(low = "white", high = "red", midpoint = 0.8) + 
  geom_tile(data = recomb_mixture_test, aes(x = lp1, y = lp2, fill = frac_on_target)) +
  geom_text(data = recomb_mixture_test, aes(x = lp1, y = lp2, label = round(frac_on_target,2)), size = 3)
ggsave(file = "plots/Double_LP_heatmap.pdf", Double_LP_heatmap, height = 2, width = 3.3)
Double_LP_heatmap
```










### UNRELATED - HOW ACQUISITION SPEED AFFECTS PRECISION OF DATA

```{r Seeing how acquisition speed affects data precision}

## Cell densities correspond to 1.66e6 for "dense", 5.25e5 for "medium", and 1.66e5 for "sparse"
opt1 <- read.csv(file = "data/flow/201106/1_dense_500.csv", header = T) %>% mutate(density = "dense", speed = 500)
opt2 <- read.csv(file = "data/flow/201106/2_dense_200.csv", header = T) %>% mutate(density = "dense", speed = 200)
opt3 <- read.csv(file = "data/flow/201106/3_dense_100.csv", header = T) %>% mutate(density = "dense", speed = 100)
opt4 <- read.csv(file = "data/flow/201106/4_med_500.csv", header = T) %>% mutate(density = "medium", speed = 500)
opt5 <- read.csv(file = "data/flow/201106/5_med_200.csv", header = T) %>% mutate(density = "medium", speed = 200)
opt6 <- read.csv(file = "data/flow/201106/6_med_100.csv", header = T) %>% mutate(density = "medium", speed = 100)
opt7 <- read.csv(file = "data/flow/201106/7_sparse_500.csv", header = T) %>% mutate(density = "sparse", speed = 500)
opt8 <- read.csv(file = "data/flow/201106/8_sparse_200.csv", header = T) %>% mutate(density = "sparse", speed = 200)
opt9 <- read.csv(file = "data/flow/201106/9_sparse_100.csv", header = T) %>% mutate(density = "sparse", speed = 100)
opt_comb <- rbind(opt1, opt2, opt3, opt4, opt5, opt6, opt7, opt8, opt9)
opt_comb$density <- as.factor(opt_comb$density)
opt_comb$speed <- as.factor(opt_comb$speed)

density_acquisition_speed_plot_1 <- ggplot() + theme_bw() + scale_x_log10() + labs(x = "BFP fluoresence", y = "Plots are different cell densities", color = "Speed") + 
  geom_density(data = opt_comb, aes(x = VL1.A, color = speed), adjust = 1/3) + facet_grid(rows =  vars(density))
ggsave(file = "plots/density_acquisition_speed_plot_b1.pdf", density_acquisition_speed_plot_1, height = 4, width = 4)

density_acquisition_speed_plot_2 <- ggplot() + theme_bw() + scale_x_log10() + labs(x = "BFP fluoresence", y = "Plots are different acquisition speeds", color = "Cell density") + 
  geom_density(data = opt_comb, aes(x = VL1.A, color = density), adjust = 1/3) + facet_grid(rows =  vars(speed))
ggsave(file = "plots/density_acquisition_speed_plot_b2.pdf", density_acquisition_speed_plot_2, height = 4, width = 4)

density_acquisition_speed_plot_1 <- ggplot() + theme_bw() + scale_x_log10() + labs(x = "BFP fluoresence", y = "Plots are different cell densities", color = "Speed") + 
  geom_density(data = opt_comb, aes(x = BL1.A, color = speed), adjust = 1/3) + facet_grid(rows =  vars(density))
ggsave(file = "plots/density_acquisition_speed_plot_g1.pdf", density_acquisition_speed_plot_1, height = 4, width = 4)

density_acquisition_speed_plot_2 <- ggplot() + theme_bw() + scale_x_log10() + labs(x = "BFP fluoresence", y = "Plots are different acquisition speeds", color = "Cell density") + 
  geom_density(data = opt_comb, aes(x = BL1.A, color = density), adjust = 1/3) + facet_grid(rows =  vars(speed))
ggsave(file = "plots/density_acquisition_speed_plot_g2.pdf", density_acquisition_speed_plot_2, height = 4, width = 4)

density_acquisition_speed_plot_1 <- ggplot() + theme_bw() + scale_x_log10() + labs(x = "BFP fluoresence", y = "Plots are different cell densities", color = "Speed") + 
  geom_density(data = opt_comb, aes(x = YL2.A, color = speed), adjust = 1/3) + facet_grid(rows =  vars(density))
ggsave(file = "plots/density_acquisition_speed_plot_y1.pdf", density_acquisition_speed_plot_1, height = 4, width = 4)

density_acquisition_speed_plot_2 <- ggplot() + theme_bw() + scale_x_log10() + labs(x = "BFP fluoresence", y = "Plots are different acquisition speeds", color = "Cell density") + 
  geom_density(data = opt_comb, aes(x = YL2.A, color = density), adjust = 1/3) + facet_grid(rows =  vars(speed))
ggsave(file = "plots/density_acquisition_speed_plot_y2.pdf", density_acquisition_speed_plot_2, height = 4, width = 4)
```

```{r}






```

















## SCRATCH

```{r Double_landing_pad}
library(reshape)
library(tidyverse)

double_landing_pad <- read.csv(file = "data/Double_landing_pad_201001.csv") %>% mutate(concat = paste(clone,condition,sep="_"))
double_landing_pad2 <- double_landing_pad[,!c(colnames(double_landing_pad) %in% c("clone","condition","singlets"))]
double_landing_pad2_melt <- melt(double_landing_pad2, id = "concat")

ggplot() + geom_bar(data = double_landing_pad2_melt, aes(x = concat, y = value, fill = variable), stat = "identity", position_dodge(width = 0.5))
```


```{r Bringing umap into this}
library(uwot)

combined_data <- rbind(c11_none[1:2000,], c11_recomb[1:2000,], c11_ap1903[1:2000,])

ggplot() + theme_bw() + scale_x_log10() + scale_y_log10() + geom_point(data = combined_data, aes(x = red, y = nir))

combined_data$zir <- combined_data$nir - combined_data$red * 0.01
ggplot() + theme_bw() + scale_x_log10() + scale_y_log10() + geom_point(data = combined_data, aes(x = red, y = zir))

combined_for_umap <- combined_data[,c("blu","grn","red","zir")]

umap <- umap(combined_for_umap, n_neighbors = 6, min_dist = 0.5, verbose = TRUE, n_threads = 8)
umap2 <- data.frame(umap)
#ggplot() + geom_point(data = ortholog_umap2, aes(x = X1, y = X2), alpha = 0.2, size = 5)

umap2$treatment <- combined_data[,"treatment"]

library(ggrepel)
umap_clustering <- ggplot() + theme_bw() + theme(panel.grid.minor = element_blank()) + 
  scale_color_manual(values = custom_color_scale) +
  labs(x = "Dimension 1", y = "Dimension 2") +
  geom_point(data = umap2 %>%
         arrange(treatment), aes(x = X1, y = X2, color = treatment), alpha = 0.1, size = 5)
umap_clustering
```








